src/DynamoUtilities/SmartObservableCollection.cs

```diff
@@ -133,7 +135,15 @@ protected override void ClearItems()
         {
             if (Items.Count > 0)
             {
-                base.ClearItems();
+                CheckReentrancy();
```

Comment:
won't this throw?

Reply:
Yes...but only if you use the BlockReentrancy API beforehand

Reply:
so then what is the goal of this call?

Reply:
I am overriding the base class method.
The base class (ObservableCollection) method supports CheckReentrancy.
I decided to keep supporting it in the overridden method  (maybe someone will actually use it later on)...

Reply:
👍 thanks for the explanation - makes sense.

Reply:
CheckReentrancy is designed to throw if someone tries to modify the Collection while it is still being modified

Reply:
reverted change